src/Discord.Net.WebSocket/Entities/Roles/SocketRole.cs

```diff
@@ -25,8 +24,8 @@ public class SocketRole : SocketEntity<ulong>, IRole
         public DateTimeOffset CreatedAt => SnowflakeUtils.FromSnowflake(Id);
         public bool IsEveryone => Id == Guild.Id;
         public string Mention => MentionUtils.MentionRole(Id);
-        public IReadOnlyCollection<SocketGuildUser> Members 
-            => Guild.Users.Where(x => x.Roles.Any(r => r.Id == Id)).ToImmutableArray();
+        public IEnumerable<SocketGuildUser> Members 
```

Comment:
Not to mention that you can't mutate the source collection from a LINQ query, anyway.